src/data/nat/choose.lean

```diff
@@ -100,8 +101,161 @@ theorem add_pow [comm_semiring α] (x y : α) (n : ℕ) :
 (commute.all x y).add_pow n
 
 /-- The sum of entries in a row of Pascal's triangle -/
-theorem sum_range_choose (n : ℕ) : 
+theorem sum_range_choose (n : ℕ) :
   ∑ m in range (n + 1), choose n m = 2 ^ n :=
 by simpa using (add_pow 1 1 n).symm
 
+/-
+Facts about more specific binomial coefficients and their sums
+-/
+
+private lemma twice_is_double (a : nat) : (a + a) = 2 * a := nat.bit0_val a
+
+private lemma double_succ (a : nat) : 2 * a.succ = 2 * a + 2 := by ring
+
+private lemma succ_not_le {m : nat} (t : m.succ ≤ m) : false :=
+begin
+  induction m, linarith,
+  apply m_ih,
+  exact nat.lt_succ_iff.mp t,
+end
+
+lemma sum_range_split_halfway
+  (f : nat → nat)
+  (m : nat)
+  : ∑ i in range (2 * m + 2), f i = (∑ i in range (m + 1), f i) + ∑ i in (Ico (m + 1) (2 * m + 2)), f i
+  :=
+begin
+  induction m with m,
+  {simp, refl},
+  {
+    rw sum_range_succ f (2 * (m + 1) + 1),
+    rw sum_range_succ f (2 * (m + 1)),
+    have s : 2 * m + 2 = 2 * (m + 1), refl,
+    rw s at m_ih,
+    rw m_ih,
+    clear m_ih,
+    rw sum_range_succ f (m + 1),
+    simp,
+    rw add_comm (∑ i in range (m + 1), f i) (∑ i in Ico (m + 2) (2 * (m + 1) + 2), f i),
+    rw <- add_assoc (f (m + 1)) _ _,
+    rw (@sum_eq_sum_Ico_succ_bot _ _ (m + 1) (2 * (m + 1) + 2) ( by linarith ) f).symm,
+    rw @sum_Ico_succ_top _ _ m.succ (2 * m.succ + 1) ( by linarith ) f,
+    rw @sum_Ico_succ_top _ _ m.succ (2 * m.succ) ( by linarith ) f,
+    ring,
+  }
+end
+
+-- This lemma exists only so that we can instantiate it with `i = m + 1`.
+private lemma reflect_sum_lemma
+  (i : nat)
```

Comment:
Idem dito, please "squeeze" the assumptions onto 1 or 2 lines.
`:` and `:=` go at the end of the lines.

Reply:
OK - horizontal space is very much at a premium given the existence of the proof-assistant pane, but I'm happy to do a linting pass before raising a PR.

Reply:
The mathlib standard width, which should be in the vscode default settings as a vertical line, is 100 characters. This is a compromise between the smaller width that I use normally and the much larger width that Scott prefers with his big widescreen monitors, but on my screen I can still fit 100 characters if the text takes up about 60% of the screen width, or down to 50% if you minimize the left sidebar (learn to use ctrl-B).